applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h

```diff
@@ -368,7 +371,13 @@ namespace MPMSearchElementUtility
                     bool is_found = SearchStructure.FindPointOnMesh(xg[0], N, pelem, result_begin, MaxNumberOfResults, Tolerance);
 
                     if (is_found == true) {
-                        condition_itr->GetGeometry() = pelem->GetGeometry();
+                        auto p_quadrature_point_geometry = condition_itr->pGetGeometry();
```

Comment:
Is that for lines in 2D?

In either scenario, you would need to update the tangents, too. I think we need to do this before the reset, though.

Reply:
at the moment those conditions are only points and if needed the normal vectors are updated from outside (if a contact condition is considered). If I understand correctly you are suggesting to update the normal/tangents according to the displacement of the background grid? I think that should be done in a differnt pr.

Reply:
Ok. I'm fine with that.

In either way, that was blocked for a long time. But now it is merged, so we can address that as soon as we need it.